### ISSUE TITLE
large regression in formula transformation 

### ISSUE THREAD

**Issue by kohlhase** [KWARC/sTeX#190]
Houston, we have  a problem. It seems that my verdict that Bruce's revamp of Math was OK was premature and that formula transformation in sTeX has regressed badly. 

In SMGloM ```latexml``` silently generates nothing (the simplest example is ```.../smglom/source/equal.en.tex``` which generates ```(written ),``` from ```(written $\eq{a,b}$),```. 

The absence of errors is why I thought that things were OK. 

Maybe more telling is ```sTeX/example/paper``` (you do not need a lmh installation for this; just a working latexml) where 
```
 $\fun{f}\RealNumbers\RealNumbers$
```
generates
```
<om:OMOBJ>
	  <om:OMA>
	    <om:OMS cd="arithmetics" name="multiplication"/>
	    <om:OMV name="\fun"/>
	    <om:OMV name="&#x1D453;"/>
	    <om:OMV name="\RealNumbers"/>
	    <om:OMV name="\RealNumbers"/>
	  </om:OMA>
	</om:OMOBJ>
```
cc: @angerhang @m-iancu

**Comment by dginev** [comment]
Generating omdoc is already doing too much to debug the regression btw, generating the LaTeXML XML and inspecting that is a lot more useful. I am still sad for the move to lmh, since there seems to be no easy way to do that.

I ended up running:

> latexmlc --path=../../sty/omdoc --path=../../sty/sref --path=../../sty/hwexam --path=../../sty/smglom --path=../../sty/cmath --path=../../sty/cmathml --path=../../sty/cnx --path=../../sty/dcm --path=../../sty/make --path=../../sty/mathhub --path=../../sty/metakeys --path=../../sty/etc --path=../../sty/metakeys --path=../../sty/mikoslides --path=../../sty/mmtlatex --path=../../sty/modules --path=../../sty/omdoc --path=../../sty/omtext --path=../../sty/owl2onto --path=../../sty/physml --path=../../sty/presentation --path=../../sty/problem --path=../../sty/rdfmeta --path=../../sty/reqdoc --path=../../sty/smglom --path=../../sty/smultiling --path=../../sty/sproof --path=../../sty/sref --path=../../sty/statements --path=../../sty/stex --path=../../sty/structview --path=../../sty/tikzinput --path=../../sty/workaddress --path=../../schema --path=../../xsl --preload=../background/pre.tex --destination=differentiable.xml differentiable.tex

which for me is decidedly not error-free:

> Conversion complete: 6 errors; 5 undefined macros[\inset, \rfrac, \RealNumbers, \fun, \absval].
Status:conversion:2
6 errors; 5 undefined macros[\inset, \rfrac, \RealNumbers, \fun, \absval]

I am also getting these errors when I run ```make differentiable.omdoc```. 

If I am seeing this right, ```\importmodule``` itself doesn't work anymore, but this is only for my local installation - I could be missing some setup step (which makes this harder than it should be)

**Comment by kohlhase** [comment]
Oh, I see that my fixes to differentiable.tex and friends are on the other branch, so please hold off on this for the moment. I have to fix this later today.

**Comment by kohlhase** [comment]
I have updated the files, they are OK from the LaTeX point of view. And they silently fail on \symdef-defined macros as well. At least the behavior is consistent now.

**Comment by kohlhase** [comment]
I have asked @cmaeder  to revert the merge of Bruce's changes on the ```master``` branch of KWARC/LaTeXML and move them to a new ```dev``` branch, so that sTeX becomes usable again on MathHub.info. Then we can sort out the merge problems in peace on ```dev```. 
I suspect that this will reopen the issue #151 (the statement that this is fixed was probably only because no formulae are built any more). We will see.

**Comment by kohlhase** [comment]
hmmmm, @cmaeder has moved Bruce's changes to the ```dev``` branch, and still we do not get any OpenMath. So something else must have gotten lost (at least we kwow who is not to blame).

**Comment by kohlhase** [comment]
I am quite stymied about what to expect here I would think that the problem must be somewhere in ```modules.sty.ltxml```, but there were no changes even near the symdef  stuff. I just moved the mh and views stuff out and that was it.

**Comment by kohlhase** [comment]
@angerhang could you please go through the commits in the master branch to see which commit broke the functionality?

**Comment by dginev** [comment]
So there seems to be something fundamentally different between the example and the implementation of ```\importmodule``` that I am looking at.

Your example has:
```latex
\importmodule[continuous]{continuous}
```

But the LaTeXML binding expects a keyval as the optional argument, where it will only load an sms file if the ```load=``` key is given. Did the spec of importmodule diverge from the binding?

That would explain the size of the regression.

**Comment by dginev** [comment]
Ignore my last comment, it seems my sTeX copy wasn't clean and didn't update correctly.

**Comment by dginev** [comment]
I dug around for a while and am now seeing the symdef-based formulas entirely gone from the final produced XML (already before post-processing). I'll dig a bit more later today.

**Comment by dginev** [comment]
Regression was introduced at: b2c7dc92d6349c9fc360bdcf29fe1a2962288f90 

Left comment there.

**Comment by dginev** [comment]
Not sure if you have an issue for the regression with the for and about attributes of definitions, I fixed that here: b48d208e2f45b13e6979f45010a2c93552950c42

We had the internal representation of LaTeXML objects make it to the stringified attributes, which is usually solved by a strategic placements of ```ToString```

**Comment by angerhang** [comment]
The syncing is done : >
